### PR TITLE
IKC-354 sidebar

### PR DESCRIPTION
**Pull request by pbelke**

### CODE REVIEW

File: kouncil-frontend/apps/kouncil/src/app/sidebar/sidebar.component.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            {{ (currentState$ | async) ? 'dock_to_right' : 'dock_to_left' }}
+          </mat-icon>
+        </button>
+      </div>
+    </div>
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  styleUrls: ['./sidebar.component.scss']
+})
+export class SidebarComponent {
```
Comment (juris-97) [inline kouncil-frontend/apps/kouncil/src/app/sidebar/sidebar.component.ts:46]: The padding and margins of the topic list near the sidebar do not look good to me (see screenshot below). Maybe we can change the position of the "Create topic" button or just add a "+" button somewhere, so there will not be so much blank space above.
image

File: kouncil-frontend/apps/kouncil/src/app/login/login.component.scss [hunk 2]
```diff
@@ -1,11 +1,15 @@
 @import '../../../../../apps/kouncil/src/styles/palette';
```
Comment (juris-97) [inline kouncil-frontend/apps/kouncil/src/app/login/login.component.scss:1]: just @import '../../styles/palette'; ?

File: kouncil-frontend/apps/kouncil/src/app/sidebar/sidebar.service.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  providedIn: 'root',
+})
+export class SidebarService {
+
+  currentState: string = 'opened';
+
+  private opened$: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(this.checkIfOpened());
+
+  changeState(): void {
+    if (this.currentState === 'opened') {
```
Comment (juris-97) [inline kouncil-frontend/apps/kouncil/src/app/sidebar/sidebar.service.ts:14]: Maybe it's better to create a dedicated enum for opened/closed values? SidebarState or something

File: kouncil-frontend/apps/kouncil/src/app/topic/toolbar/topic-toolbar.component.ts [hunk 4]
```diff
@@ -51,9 +51,9 @@ export enum LiveUpdateState {
       </button>
     </div>
   `,
-  styleUrls: ['./toolbar.component.scss']
+  styleUrls: ['./topic-toolbar.component.scss']
```
Comment (juris-97) [inline kouncil-frontend/apps/kouncil/src/app/topic/toolbar/topic-toolbar.component.ts:54]: Also the selector name can be changed to app-topic-toolbar

File: kouncil-frontend/apps/kouncil/src/app/topic/toolbar/topic-toolbar.component.ts [hunk 5]
```diff
@@ -51,9 +51,9 @@ export enum LiveUpdateState {
...
 })
-export class ToolbarComponent {
+export class TopicToolbarComponent {
```
Comment (juris-97) [inline kouncil-frontend/apps/kouncil/src/app/topic/toolbar/topic-toolbar.component.ts:56]: Also the selector name can be changed to app-topic-toolbar

File: kouncil-frontend/apps/kouncil/src/app/sidebar/sidebar.component.ts (hunk 1 above)
Comment (pbelke) [reply kouncil-frontend/apps/kouncil/src/app/sidebar/sidebar.component.ts:47]: As discussed, we will think about some additional functionality that will make this bar more appealing

**Review by juris-97 — APPROVED** [review]

**Pull request merged**